docs/model-monitoring/applications.md

```diff
@@ -0,0 +1,114 @@
+# Writing a Model Monitoring Application
+
+## Basics
+
+First, create a Python module and import the API objects:
+
+```py
+from mlrun.model_monitoring.applications import (
+    ModelMonitoringApplicationBase,
+    ModelMonitoringApplicationMetric,
+    ModelMonitoringApplicationResult,
+    MonitoringApplicationContext,
+)
+from mlrun.common.schemas.model_monitoring import ResultKindApp, ResultStatusApp
+```
+
+Then, write the application itself by inheriting from the `ModelMonitoringApplicationBase` base class.
+You have to implement the `do_tracking` method.
+Here is a "dummy" app that returns a constant result for each monitoring window:
+
+```py
+class ServingMonitoring(ModelMonitoringApplicationBase):
+    def do_tracking(
+        self, context: MonitoringApplicationContext
+    ) -> ModelMonitoringApplicationResult:
+        return ModelMonitoringApplicationResult(
+            name="dummy-res-just-for-demo",
+            value=0,
+            status=ResultStatusApp.irrelevant,
+            kind=ResultKindApp.mm_app_anomaly,
+        )
+```
+
+The `do_tracking` method of the application is called for each monitoring window, and returns a list
+or results `ModelMonitoringApplicationResult` and metrics `ModelMonitoringApplicationMetric`, or just one
+result, as in the example above.
+
+```{versionchanged} 1.7.0 ModelMonitoringApplicationBase. Please update your application accordingly.
+```
+
+## Using the Application Context
+
+The `context` argument is a `MonitoringApplicationContext` object.
+It includes the current window data as a pandas data-frame: `context.sample_df`.
+The reference and current data is also available in raw format as `context.feature_stats`
+and `context.sample_df_stats`, respectively.
+
+The `context` provides also attributes and methods to log application messages or artifacts.
+
+Logging a debug message:
+
+```py
+context.logger.debug(
+    "Logging the current data of a specific endpoint",
+    sample_df=context.sample_df.to_json(),
+    endpoint_id=context.endpoint_id,
+)
+```
+
+Logging an artifact:
+
+```py
+context.log_artifact(
+    item=f"num_events_last_monitoring_window_{context.endpoint_id}",
+    body=f"Number of events in the window: {len(context.sample_df)}",
+)
+```
+
+```{caution}
+Do not log new artifacts in every standard run of your application, as the artifact's content is
+currently saved in memory and will aggregate with the number of calls to the application.
+
```

Comment:
```suggestion
Since each new artifact is saved in the artifact-manager store (in memory), it is not recommended to store a new artifact on each application run. Instead you may:

1. Override the artifact of previous run by using the same key.
2. Save artifacts with a unique key in special occasions, e.g., when a drift is detected.
```